File: app/src/main/java/com/metahumanz/pacilread/tts/TtsPlaybackSnapshot.java
package com.metahumanz.pacilread.tts;

public final class TtsPlaybackSnapshot {
    public static final String STATE_STOPPED = "stopped";
    public static final String STATE_PLAYING = "playing";
    public static final String STATE_PAUSED = "paused";
    public static final String STATE_LOADING = "loading";

    public final long bookId;
    public final String bookTitle;
    public final String chapterTitle;
    public final int chapterIndex;
    public final int sentenceStart;
    public final int sentenceEnd;
    public final String state;
    public final long sleepDeadlineElapsed;

    public TtsPlaybackSnapshot(long bookId, String bookTitle, String chapterTitle,
                               int chapterIndex, int sentenceStart, int sentenceEnd,
                               String state, long sleepDeadlineElapsed) {
        this.bookId = bookId;
        this.bookTitle = bookTitle == null ? "" : bookTitle;
        this.chapterTitle = chapterTitle == null ? "" : chapterTitle;
        this.chapterIndex = chapterIndex;
        this.sentenceStart = sentenceStart;
        this.sentenceEnd = sentenceEnd;
        this.state = state == null ? STATE_STOPPED : state;
        this.sleepDeadlineElapsed = sleepDeadlineElapsed;
    }

    public boolean isActive() {
        return STATE_PLAYING.equals(state) || STATE_PAUSED.equals(state) || STATE_LOADING.equals(state);
    }

    public boolean isPaused() {
        return STATE_PAUSED.equals(state);
    }
}

File: app/src/main/kotlin/com/metahumanz/pacilread/tts/TtsPlaybackSnapshot.kt
package com.metahumanz.pacilread.tts

class TtsPlaybackSnapshot(
    @JvmField val bookId: Long,
    bookTitle: String?,
    chapterTitle: String?,
    @JvmField val chapterIndex: Int,
    @JvmField val sentenceStart: Int,
    @JvmField val sentenceEnd: Int,
    state: String?,
    @JvmField val sleepDeadlineElapsed: Long,
) {
    @JvmField val bookTitle: String = bookTitle ?: ""
    @JvmField val chapterTitle: String = chapterTitle ?: ""
    @JvmField val state: String = state ?: STATE_STOPPED

    fun isActive(): Boolean = state == STATE_PLAYING || state == STATE_PAUSED || state == STATE_LOADING

    fun isPaused(): Boolean = state == STATE_PAUSED

    companion object {
        const val STATE_STOPPED = "stopped"
        const val STATE_PLAYING = "playing"
        const val STATE_PAUSED = "paused"
        const val STATE_LOADING = "loading"
    }
}
